tests/Unit/NumericalAlgorithms/Integration/Test_GslQuadAdaptive.cpp

```diff
@@ -0,0 +1,111 @@
+
+#include "Framework/TestingFramework.hpp"
+
+#include <cmath>
+#include <cstddef>
+
+#include "Framework/TestHelpers.hpp"
+#include "NumericalAlgorithms/Integration/GslQuadAdaptive.hpp"
+
+namespace {
+/// [integrated_function]
+double gaussian(double x, double mean, double factor) {
+  return 2. * factor / sqrt(M_PI) * exp(-square(x - mean));
+}
+
+/// [integrated_function]
+double integrable_singularity(double x, double factor) {
+  return factor * cos(sqrt(abs(x))) / sqrt(abs(x));
+}
+
+SPECTRE_TEST_CASE("Unit.Numerical.Integration.GslQuadAdaptive",
+                  "[Unit][NumericalAlgorithms]") {
+  {
+    INFO("StandardGaussKronrod");
+    // Construct the integration and give an example
+    /// [integration_example]
+    integration::GslQuadAdaptive<
+        integration::GslIntegralType::StandardGaussKronrod>
+        integration{20};
+    const double mean = 5.;
+    const double factor = 2.0;
+    const double lower_boundary = -4.;
+    const double upper_boundary = 10.;
+    const auto result = integration(
+        [&mean, &factor](const double x) { return gaussian(x, mean, factor); },
+        lower_boundary, upper_boundary, 1.e-10, 4);
+    /// [integration_example]
+    CHECK(result == approx(factor * erf(upper_boundary - mean) -
+                           factor * erf(lower_boundary - mean)));
+  }
+
+  {
+    INFO("InfiniteInterval");
+    integration::GslQuadAdaptive<integration::GslIntegralType::InfiniteInterval>
+        integration{20};
+    const double mean = 5.;
+    const double factor = 2.0;
+    const auto result = integration(
+        [&mean, &factor](const double x) { return gaussian(x, mean, factor); },
+        1.e-10);
+    CHECK(result == approx(2. * factor));
+  }
+
+  {
+    INFO("UpperBoundaryInfinite");
+    integration::GslQuadAdaptive<
+        integration::GslIntegralType::UpperBoundaryInfinite>
+        integration{20};
+    const double mean = 5.;
+    const double factor = 2.0;
+    const double lower_boundary = -4.;
+    auto result = integration(
+        [&mean, &factor](const double x) { return gaussian(x, mean, factor); },
+        lower_boundary, 1.e-10);
+    CHECK(result == approx(factor * (1 - erf(lower_boundary - mean))));
+  }
+
+  {
+    INFO("LowerBoundaryInfinite");
+    integration::GslQuadAdaptive<
+        integration::GslIntegralType::LowerBoundaryInfinite>
+        integration{20};
+    const double mean = 5.;
+    const double factor = 2.0;
+    const double upper_boundary = 10.;
+    auto result = integration(
+        [&mean, &factor](const double x) { return gaussian(x, mean, factor); },
+        upper_boundary, 1.e-10);
+    CHECK(result == approx(factor * (1 + erf(upper_boundary - mean))));
+  }
+
+  {
+    INFO("IntegrableSingularitiesPresent");
+    integration::GslQuadAdaptive<
+        integration::GslIntegralType::IntegrableSingularitiesPresent>
+        integration{20};
+    const double factor = 2.0;
+    const double upper_boundary = square(M_PI / 2.);
```

Comment:
`0.5 * M_PI`

Reply:
I have changed it to `0.5 * M_PI`, but am unclear on what the benefit is?

Reply:
Good question! Here it's not so important, but multiplication is ~27x faster than division, thus multiplying be `0.5` is significantly faster than dividing by `2.0`. In a test it's not really important, but I think it's good to have the tests lead by example :) Does that make sense?